src/webview/pipeline/app/widgets/inputwidget.ts

```diff
@@ -31,18 +31,32 @@ export class InputWidget extends BaseWidget {
     this.input.placeholder = text ?? '';
     this.input.value = inputValue ?? '';
     this.input.type = number ?? 'text';
+    if (number) {
+      this.input.min = '0';
+    }
     this.input.id = inputId ?? '';
     this.input.disabled = disabledType ?? false;
     this.input.title = inputTitle ?? '';
     this.element = editorInput;
     const wrapper = createDivWithID('wrapper', wrapperId);
     wrapper.appendChild(this.input);
+    this.input.onkeydown = (event) => this.blockNegativeNumber(event, this.input);
     this.input.oninput = () => this.getValue(this.input);
     this.input.onblur = () => this.validator();
     this.input.onfocus = () => this.removeError();
     editorInput.appendChild(wrapper);
   }
 
+  blockNegativeNumber(event: KeyboardEvent, input: HTMLInputElement): boolean {
+    if (input?.id === 'size-for-pvc-create-webview') {
+      const matchInputNumber = new RegExp('[0-9]|(Backspace)');
+      if (!event.key.match(matchInputNumber)) {
```

Comment:
Everything works fine but i wanted to ask you why don't you verify the input instead of the keyboardEvent? I mean, every time someone types something, check if the input contains a positive valid number otherwise do not add the last value typed.

In the current way there are some limitations like one cannot move using right/left arrows or delete using canc or users can type numbers like 000020 which javascript convert using octals, not sure kubernetes.

Reply:
Hi, @lstocchi if we create an html input with type=number write "- or e" in it and try to access it's value via JS you will get an empty string so I use keyboardEvent or alternative, I use input type=text, use regex to allow only digits.

Reply:
Ok. I guess it can be left as it is, it's not an input that will be used so often

Reply:
> some limitations like one cannot move using right/left arrows

but I have fix right/left arrows related problem.